pinot-common/src/main/java/com/linkedin/pinot/common/metadata/segment/PartitionRangesDeserializer.java

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.common.metadata.segment;
+
+import com.linkedin.pinot.common.config.ColumnPartitionConfig;
+import java.io.IOException;
+import java.util.List;
+import org.apache.commons.lang.math.IntRange;
+import org.codehaus.jackson.JsonParser;
+import org.codehaus.jackson.map.DeserializationContext;
+import org.codehaus.jackson.map.JsonDeserializer;
+
+
+/**
+ * Custom Json de-serializer for list of IntRange's.
+ */
+public class PartitionRangesDeserializer extends JsonDeserializer<List<IntRange>> {
+
+  @Override
+  public List<IntRange> deserialize(JsonParser jsonParser, DeserializationContext context)
+      throws IOException {
+    return ColumnPartitionConfig.rangesFromString(
+        jsonParser.getText().split(ColumnPartitionConfig.PARTITION_VALUE_DELIMITER));
```

Comment:
ColumnPartitionConfig.rangesFromString() should take a single String as argument instead of a String array